presto-spi/src/main/java/io/prestosql/spi/Page.java

```diff
@@ -311,16 +317,17 @@ public Page prependColumn(Block column)
         result[0] = column;
         System.arraycopy(blocks, 0, result, 1, blocks.length);
 
-        return new Page(positionCount, result);
+        return new Page(false, positionCount, result);
     }
 
-    private void updateRetainedSize()
+    private long updateRetainedSize()
```

Comment:
undo

Reply:
The alternative is a redundant `volatile` read in `getRetainedSize()` which is probably not *that* bad but seemed worse than modifying the return type here. Would you be satisfied with a rename (eg: `updateAndReturnRetainedSize()`) or is there something specific you want to preserve by keeping this method `void`?